### PR TITLE
RUBY-2592 Convert change stream spec tests to unified test format

### PR DESCRIPTION
**Pull request by Neilshweky**
There are errors on CI only with respect to load balancers. They have been removed from this PR with a comment. I am leaving this ticket open to work out a resolution at a later time. The failure output is below.

### CODE REVIEW

File: spec/runners/unified/test.rb [hunk 1]
```diff
@@ -156,6 +156,8 @@ def create_entities
                   raise NotImplementedError, "Unknown event #{event}"
                 end
               end
+            else
+              subscriber = (@subscribers[client] ||= EventSubscriber.new)
```
Comment (p-mongo) [inline spec/runners/unified/test.rb:160]: The subscriber is now always created, right? If so its creation should be lifted out of the branches to before line 135 and done once.

**Review by p-mongo — COMMENTED** [review]
For the failing tests you can remove them from our source and leave the ticket open to work out a resolution at a later time.

**Comment by Neilshweky** [comment]
Load balancer failures: 
```
1) Change stream unified spec tests change-streams-errors.yml change stream errors on ElectionInProgress passes
    Failure/Error: raise Error::ErrorMismatch, "Expected exception but none was raised"
    
    Unified::Error::ErrorMismatch:
      Expected exception but none was raised
    # ./spec/runners/unified/test.rb:334:in `block in execute_operation'
    # ./spec/runners/unified/test.rb:362:in `use_all'
    # ./spec/runners/unified/test.rb:277:in `execute_operation'
    # ./spec/runners/unified/test.rb:272:in `block in execute_operations'
    # ./spec/runners/unified/test.rb:271:in `each'
    # ./spec/runners/unified/test.rb:271:in `execute_operations'
    # ./spec/runners/unified/test.rb:241:in `run'
    # ./spec/runners/unified.rb:89:in `block (5 levels) in define_unified_spec_tests'
    # ./spec/lite_spec_helper.rb:136:in `block (3 levels) in <top (required)>'
    # ./spec/lite_spec_helper.rb:135:in `block (2 levels) in <top (required)>'
    # ./spec/lite_spec_helper.rb:115:in `block (2 levels) in <top (required)>'
    # ./spec/support/background_thread_registry.rb:65:in `block (2 levels) in <top (required)>'

2) Change stream unified spec tests change-streams-resume-allowlist.yml change stream resumes after a network error passes
    Failure/Error: raise error
    
    Mongo::Error::OperationFailure:
      [216:ElectionInProgress]: Failing command due to 'failCommand' failpoint (on localhost:27017)
    # ./lib/mongo/operation/result.rb:364:in `raise_operation_failure'
    # ./lib/mongo/operation/result.rb:330:in `validate!'
    # ./lib/mongo/operation/shared/response_handling.rb:36:in `block (3 levels) in validate_result'
    # ./lib/mongo/operation/shared/response_handling.rb:108:in `add_server_diagnostics'
    # ./lib/mongo/operation/shared/response_handling.rb:35:in `block (2 levels) in validate_result'
    # ./lib/mongo/operation/shared/response_handling.rb:54:in `add_error_labels'
    # ./lib/mongo/operation/shared/response_handling.rb:34:in `block in validate_result'
    # ./lib/mongo/operation/shared/response_handling.rb:94:in `unpin_maybe'
    # ./lib/mongo/operation/shared/response_handling.rb:33:in `validate_result'
    # ./lib/mongo/operation/shared/executable.rb:81:in `block in execute'
    # ./lib/mongo/operation/shared/executable.rb:80:in `tap'
    # ./lib/mongo/operation/shared/executable.rb:80:in `execute'
    # ./lib/mongo/operation/shared/polymorphic_operation.rb:50:in `execute_with_connection'
    # ./lib/mongo/operation/shared/polymorphic_operation.rb:36:in `block in execute'
    # ./lib/mongo/server/connection_pool.rb:666:in `with_connection'
    # ./lib/mongo/server.rb:448:in `with_connection'
    # ./lib/mongo/operation/shared/polymorphic_operation.rb:35:in `execute'
    # ./lib/mongo/cursor.rb:485:in `execute_operation'
    # ./lib/mongo/cursor.rb:366:in `get_more'
    # ./lib/mongo/cursor.rb:221:in `try_next'
    # ./lib/mongo/collection/view/change_stream.rb:181:in `try_next'
    # ./lib/mongo/collection/view/change_stream.rb:159:in `block in each'
    # ./lib/mongo/collection/view/change_stream.rb:158:in `loop'
    # ./lib/mongo/collection/view/change_stream.rb:158:in `each'

3) Change stream unified spec tests change-streams-resume-allowlist.yml change stream resumes after CursorNotFound passes
    Failure/Error: raise Error::ConnectionCheckOutTimeout.new(msg, address: @server.address)
    
    Mongo::Error::ConnectionCheckOutTimeout:
      Timed out attempting to check out a connection from pool for localhost:27017 for connection 15 after 6.04 sec. Connections in pool: 0 available, 0 checked out, 0 pending (max size: 3)
    # ./lib/mongo/server/connection_pool.rb:411:in `block (2 levels) in check_out'
    # ./lib/mongo/server/connection_pool.rb:304:in `loop'
    # ./lib/mongo/server/connection_pool.rb:304:in `block in check_out'
    # ./lib/mongo/server/connection_pool.rb:303:in `catch'
    # ./lib/mongo/server/connection_pool.rb:303:in `check_out'
    # ./lib/mongo/server/connection_pool.rb:665:in `with_connection'
    # ./lib/mongo/server.rb:448:in `with_connection'
    # ./lib/mongo/operation/shared/polymorphic_operation.rb:35:in `execute'
    # ./lib/mongo/cursor.rb:485:in `execute_operation'
    # ./lib/mongo/cursor.rb:291:in `block in close'
    # ./lib/mongo/retryable.rb:165:in `read_with_one_retry'
    # ./lib/mongo/cursor.rb:284:in `close'
    # ./lib/mongo/collection/view/change_stream.rb:235:in `close'
    # ./lib/mongo/collection/view/change_stream.rb:195:in `rescue in try_next'
    # ./lib/mongo/collection/view/change_stream.rb:180:in `try_next'
    # ./lib/mongo/collection/view/change_stream.rb:159:in `block in each'
    # ./lib/mongo/collection/view/change_stream.rb:158:in `loop'
    # ./lib/mongo/collection/view/change_stream.rb:158:in `each'
    # ------------------
    # --- Caused by: ---
    # EOFError:
    #   end of file reached
    #   ./lib/mongo/socket.rb:307:in `read_from_socket'

4) Change stream unified spec tests change-streams-resume-errorLabels.yml change stream resumes after HostUnreachable passes
    Failure/Error: raise Error::ResultMismatch, "#{msg}: expected #{expected}, actual #{actual}"
    
    Unified::Error::ResultMismatch:
      Command names differ: expected aggregate, actual getMore
    # ./spec/runners/unified/assertions.rb:161:in `assert_eq'
    # ./spec/runners/unified/assertions.rb:146:in `assert_event_matches'
    # ./spec/runners/unified/assertions.rb:125:in `block (2 levels) in assert_events'
    # ./spec/runners/unified/assertions.rb:124:in `each'
    # ./spec/runners/unified/assertions.rb:124:in `each_with_index'
    # ./spec/runners/unified/assertions.rb:124:in `block in assert_events'
    # ./spec/runners/unified/assertions.rb:96:in `each'
    # ./spec/runners/unified/assertions.rb:96:in `assert_events'
    # ./spec/runners/unified.rb:91:in `block (5 levels) in define_unified_spec_tests'
    # ./spec/lite_spec_helper.rb:136:in `block (3 levels) in <top (required)>'
    # ./spec/lite_spec_helper.rb:135:in `block (2 levels) in <top (required)>'
    # ./spec/lite_spec_helper.rb:115:in `block (2 levels) in <top (required)>'
    # ./spec/support/background_thread_registry.rb:65:in `block (2 levels) in <top (required)>'

5) Change stream unified spec tests change-streams-resume-errorLabels.yml change stream does not resume if error does not contain ResumableChangeStreamError passes
    Failure/Error: raise Error::ErrorMismatch, "Expected exception but none was raised"
    
    Unified::Error::ErrorMismatch:
      Expected exception but none was raised
    # ./spec/runners/unified/test.rb:334:in `block in execute_operation'
    # ./spec/runners/unified/test.rb:362:in `use_all'
    # ./spec/runners/unified/test.rb:277:in `execute_operation'
    # ./spec/runners/unified/test.rb:272:in `block in execute_operations'
    # ./spec/runners/unified/test.rb:271:in `each'
    # ./spec/runners/unified/test.rb:271:in `execute_operations'
    # ./spec/runners/unified/test.rb:241:in `run'
    # ./spec/runners/unified.rb:89:in `block (5 levels) in define_unified_spec_tests'
    # ./spec/lite_spec_helper.rb:136:in `block (3 levels) in <top (required)>'
    # ./spec/lite_spec_helper.rb:135:in `block (2 levels) in <top (required)>'
    # ./spec/lite_spec_helper.rb:115:in `block (2 levels) in <top (required)>'
    # ./spec/support/background_thread_registry.rb:65:in `block (2 levels) in <top (required)>'
```

File: spec/spec_tests/unified_spec.rb [hunk 2]
```diff
@@ -6,10 +6,12 @@
 require 'runners/unified'
 
 base = "#{CURRENT_PATH}/spec_tests/data/unified"
+INVALID_UNIFIED_TESTS = Dir.glob("#{base}/invalid/**/*.yml").sort
 PASS_UNIFIED_TESTS = Dir.glob("#{base}/valid-pass/**/*.yml").sort
 FAIL_UNIFIED_TESTS = Dir.glob("#{base}/valid-fail/**/*.yml").sort
 
 describe 'Unified spec tests' do
+  define_unified_spec_tests(base, INVALID_UNIFIED_TESTS)
```
[the remark this answers is not in the document]
Comment (Neilshweky) [reply spec/spec_tests/unified_spec.rb:14]: done

**Review by p-mongo — APPROVED** [review]

**Comment by Neilshweky** [comment]
The above is for multi-lb. for single-lb only the following error occurs:
```
1) Change stream unified spec tests change-streams-resume-allowlist.yml change stream resumes after a network error passes
      Failure/Error: raise Error::ConnectionCheckOutTimeout.new(msg, address: @server.address)
      
      Mongo::Error::ConnectionCheckOutTimeout:
        Timed out attempting to check out a connection from pool for localhost:27017 for connection 12 after 6.04 sec. Connections in pool: 0 available, 0 checked out, 0 pending (max size: 3)
      # ./lib/mongo/server/connection_pool.rb:411:in `block (2 levels) in check_out'

**Pull request merged**